pkg/client/common/discovery/dynamicdiscovery/service.go

```diff
@@ -139,17 +140,28 @@ func asPeer(ctx contextAPI.Client, endpoint *discclient.Peer) (fab.Peer, bool) {
 
 	logger.Debugf("Adding endpoint [%s]", url)
 
+	var (
+		peer fab.Peer
+		err  error
+	)
+
 	peerConfig, found := ctx.EndpointConfig().PeerConfig(url)
-	if !found {
-		logger.Debugf("Peer config not found for url [%s]", url)
-		return nil, false
+	if found {
+		peer, err = ctx.InfraProvider().CreatePeerFromConfig(&fab.NetworkPeer{
+			PeerConfig: *peerConfig,
+			MSPID:      endpoint.MSPID,
+			Properties: fabdiscovery.GetProperties(endpoint),
+		})
+	} else {
+		peer, err = peerImpl.New(ctx.EndpointConfig(), peerImpl.FromPeerConfig(&fab.NetworkPeer{
+			PeerConfig: fab.PeerConfig{
```

Comment:
There are two other properties in PeerConfig:

https://github.com/hyperledger/fabric-sdk-go/blob/7bf58790be693b74f78e37393781988244b0290b/pkg/common/providers/fab/network.go#L137-L138

In the discovery case (without config), what are these values being set to? In particular, is `TLSCACert` being handled properly?

Reply:
I didn't introduce any breaking changes, nothing will change for current users at this point. The old way of selection among peers from the Endpointconfig is still present.

In the case when a peer is not found config, no TLSCA or gRPC options will be applied. Missing TLSCACert is ok because you can have your own implementation of the `EndpointConfig.TLSCACertPool()` which returns a cert pool, it's on the user to manage it. Regarding gRPC options,  at the moment they will remain empty if the peer is not present in the `EndpointConfig`.  There is no way to get them from the discovery service or from `_default` because it is a part only of the particular SDK implementation and evaluated at the startup for specified peers.  A better solution would be to add a global communication config to the `EndpointConfig` like `EndpointConfig().CommConfig()`